tests/db_utils/tests.py

```diff
@@ -21,6 +21,27 @@ def test_connection_handler_no_databases(self):
         with self.assertRaisesMessage(ImproperlyConfigured, msg):
             conns[DEFAULT_DB_ALIAS].ensure_connection()
 
+    def test_connection_handler_empty_default_database(self):
+        """Empty default database setting defaults to the dummy backend."""
+        DATABASES = {DEFAULT_DB_ALIAS: {}}
+        conns = ConnectionHandler(DATABASES)
+        self.assertEqual(conns[DEFAULT_DB_ALIAS].settings_dict['ENGINE'], 'django.db.backends.dummy')
+        msg = (
+            'settings.DATABASES is improperly configured. Please supply the '
+            'ENGINE value. Check settings documentation for more details.'
+        )
+        with self.assertRaisesMessage(ImproperlyConfigured, msg):
+            conns[DEFAULT_DB_ALIAS].ensure_connection()
```

Comment:
This test is not related with the fix, so it will be better to create a separate PR for increasing tests coverage. I would add `DATABASES = {DEFAULT_DB_ALIAS: {}}` as a next case to the `test_connection_handler_no_databases()` (probably renamed), e.g.
```python
def test_connection_handler_no_databases(self):
    for DATABASES in (
        {},  # Empty DATABASES setting.
        {DEFAULT_DB_ALIAS: {}},  # Empty default database setting.
    ):
         conns = ConnectionHandler(DATABASES)
         ...
```

Reply:
Removed the unrelated test.